stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py

```diff
@@ -446,15 +446,30 @@ def apply_datetime_filter(
         Returns:
             The filtered search object.
         """
+        datetime_search = return_date(datetime)
+
         # USE_DATETIME env var
         # True: Search by datetime, if null search by start/end datetime
         # False: Always search only by start/end datetime
         USE_DATETIME = get_bool_env("USE_DATETIME", default=True)
 
-        datetime_search = return_date(datetime)
+        result_metadata = {
+            "datetime": {
+                "gte": datetime_search.get("gte") if USE_DATETIME else None,
+                "lte": datetime_search.get("lte") if USE_DATETIME else None,
+            },
+            "start_datetime": {
+                "gte": datetime_search.get("gte") if not USE_DATETIME else None,
+                "lte": None,
+            },
+            "end_datetime": {
+                "gte": None,
+                "lte": datetime_search.get("lte") if not USE_DATETIME else None,
+            },
+        }
```

Comment:
This added code complicates the core database logic by tightly coupling it to a specific indexing strategy. Please move this calculation into the IndexSelector (the actual consumer) to keep the core method focused solely on query construction.